docs: db migration (#265)
logto-io/docs: docs/docs/recipes/deployment/README.md

If we don't mention it in the changelog, you can feel free to upgrade Logto without changing your code or database schema. Our API follows [semver](https://semver.org/).

> ### Database migration

Reviewer: should be "alternation"? then we can separate from "migrate from other projects"

Author: Sure, then I'll also change `migration` from code later.

> If a schema change is inevitable, we will provide a migration script. You don't need to care about it, simply run "pnpm migration-deploy" to upgrade your database schema.

Reviewer 2: ```suggestion
If a schema change is inevitable, we will provide a migration script. Simply run "pnpm migration-deploy" with ease to upgrade your database schema.
```
